src/cmd/update.js

```diff
@@ -43,8 +48,10 @@ module.exports = class UpdateCommand extends CLICommandBase {
 		const modulesToFlash = filterModulesToFlash({ modules: deviceOsModules, platformId: device.platformId, allowAll: true });
 		await maintainDeviceProtection({ modules: modulesToFlash, device });
 		const flashSteps = await createFlashSteps({ modules: modulesToFlash, isInDfuMode: device.isInDfuMode , platformId: device.platformId });
-		await flashFiles({ device, flashSteps, ui: this.ui });
+		device = await flashFiles({ device, flashSteps, ui: this.ui });
+
 		this.ui.write('Update success!');
```

Comment:
Do you need to reopen the device here like after the other invocations of `flashFiles`